version: "3.3"

services:
    db-dev:
        image: mariadb:11.2.2-jammy
        restart: always
        environment:
            MARIADB_DATABASE: ${DB_DATABASE}
            MARIADB_USER: ${DB_USER}
            MARIADB_PASSWORD: ${DB_PASSWORD}
            MARIADB_ROOT_PASSWORD: ${DB_PASSWORD}
        networks:
            - default
        ports:
            - "3307:3306"
        volumes:
            - ./db:/var/lib/mysql
        healthcheck:
            test:
                [
                    "CMD",
                    "healthcheck.sh",
                    "--su-mysql",
                    "--connect",
                    "--innodb_initialized",
                ]
            interval: 1s
            timeout: 20s
            retries: 30

    api-dev:
        build:
            context: .
            dockerfile: Dockerfile
        ports:
            - "81:3000"
        volumes:
            - .:/app
        environment:
            NODE_ENV: development
            PORT: 3000
            DB_HOST: "db-dev"
            DB_DATABASE: ${DB_DATABASE}
            DB_USER: ${DB_USER}
            DB_PASSWORD: ${DB_PASSWORD}
                - "traefik.enable=true"
                - "traefik.http.routers.api-dev.rule=Host(`dev.${HOST}`)"
                - "traefik.http.routers.api-dev.entrypoints=websecure"
                - "traefik.http.routers.api-dev.tls.certresolver=myresolver"
                - "traefik.http.services.api-dev.loadbalancer.server.port=3000"
                - "traefik.http.routers.api-dev-insecure.entrypoints=web"
                - "traefik.http.middlewares.api-dev-insecure.redirectscheme.scheme=https"
                - "traefik.docker.network=traefik"
        networks:
            - default
            - traefik
        depends_on:
            db-dev:
                condition: service_healthy

networks:
    traefik:
        driver: bridge
        external: true
